### ISSUE TITLE
Assets installable only after manual web fetch

### ISSUE THREAD

**Issue by scaytrase** [hiqdev/asset-packagist#11]
I'm trying to install update some assets via npm-asset notation. I.e I take `nestedSortable` package from NPM. I add `"npm-asset/nestedSortable":"*"` to the composer.json, do `composer update` and receive 

```
  Problem 1
    - The requested package npm-asset/nestedsortable could not be found in any version, there may be a typo in the package name.
```

Packagist repository is added according to manual.

Then I go to https://asset-packagist.org/site/search?query=npm-asset%2FnestedSortable to check the package exist. The site fetch me versions  successfully. After this I do `composer update` again.

And all works (no code or composer changes)

```
  - Installing npm-asset/nestedsortable (2.0.0.0)
    Downloading: Connecting...    Downloading: 100%         
```

And repeats when I take other package.

Looks like some packagist caching problem

**Comment by hiqsol** [comment]
That's not a problem. This is how it works at the moment.

Composer is arranged in the way that new packages can not be added to the asset packagist on demand. All packages have to be fetched to asset packagist before they can be used.
We have already added 1000+ most popular Bower packages and 1000+ most popular NPM packages
to the asset packagist. And we have the web interface for anybody could add all the packages they need. 

We thought about forced adding of all of the available Bower and NPM packages...
But considering total number of NPM (~300K) and Bower (~50K) packages it seems doubtful.
Composer is designed in the way that it downloads full list of packages from repository. Composer has reasonable caching but still it will have to download full list at least once on the first use.

So far for the moment you need to fetch all the packages you want using the web interface.
But you need to do it just once. I hope it is not a big problem.

**Comment by scaytrase** [comment]
I think this should be mentioned on `usage` section then. Advice to check the package on the web first, try to add after.

**Comment by hiqsol** [comment]
Actually, it is mentioned on the index page.
But I agree the description needs to be improved.

**Comment by SilverFire** [comment]
Currently is it's not possible to pass missing package error from local composer to asset-packagist automatically, so the problem can not be solved on our side.

We've loaded 7900+ TOP popular packages and their dependencies from Bower and NPM. We update all packages we have on a regular basis.

**Comment by SilverFire** [comment]
Documentation might be improved though :)

**Comment by SilverFire** [comment]
Done ec7bbcd57978f9979c2d23ba1e958e2cf97b2c38

**Comment by dmitry-kulikov** [comment]
Today I tried to move from fxp/composer-asset-plugin to Asset Packagist. To have clear experiment I removed vendor directory and composer.lock. After `composer update` I noticed that there was installed npm-asset/jsoneditor version 5.0.1, then during review of changes in composer.lock before commit (I keep composer.lock in repository) I noticed that as a result npm-asset/jsoneditor was downgraded from 5.7.0 to 5.0.1. My composer.json contained `"npm-asset/jsoneditor": "5.*"`, I tried to require 5.7 as follow: `"npm-asset/jsoneditor": "5.7.*"` and got
```
Your requirements could not be resolved to an installable set of packages.

  Problem 1
    - npm-asset/ajv 4.11.6 requires npm-asset/co >=4.6.0,<5.0.0 -> no matching package found.
    - npm-asset/jsoneditor 5.7.0 requires npm-asset/ajv 4.11.6 -> satisfiable by npm-asset/ajv[4.11.6].
    - Installation request for npm-asset/jsoneditor 5.7.* -> satisfiable by npm-asset/jsoneditor[5.7.0].
```
it happened because new version of npm-asset/jsoneditor requires packages which Asset Packagist still doesn't know. I can add manually all needed packages to Asset Packagist, but usually people doesn't monitor subdependencies, because relies on package maintainer (I mean jsoneditor) and semantic versioning. But without needed packages for new version composer will try to install old ones and may succeed. This may be significant problem in projects with many dependecies, because you never know do you really have latest versions or no until check manually. Also many developers may don't notice problem and stuck on old version with bugs.

**Comment by hiqsol** [comment]
@dmitry-kulikov your comment is worth of separate issue! :) Don't hesitate to open new issue next time.

Actually we have auto updating process but unfortunately it was broken.
Fixed. Up and running.
